stablehlo/dialect/TypeInference.h

```diff
@@ -147,7 +147,9 @@ LogicalResult inferClampOp(
     SmallVectorImpl<ShapedTypeComponents>& inferredReturnShapes);
 
 LogicalResult inferCompareOp(
-    MLIRContext* context, std::optional<Location>, Value lhs,
+    MLIRContext* context, std::optional<Location>, bool compareTypeHasValue,
+    bool isNoType, bool isFloat, bool isTotalOrder, bool isUnsigned,
```

Comment:
Other similar flags in this file are called `isFieldNameFieldValue`, e.g. `isCompareTypeNoType` etc. Let's rename accordingly?